frontend/src/components/CreateDataSourceExample.vue

```diff
@@ -0,0 +1,201 @@
+        <a
+          href="https://docs.bytebase.com/install/docker#start-a-local-mysql-server-for-testing"
+          target="_blank"
+          class="normal-link"
+        >
+          {{ $t("common.detailed-guide") }}
+        </a>
+      </template·>
+      <template v-else-if="props.engineType == 'CLICKHOUSE'">
+        <i18n-t
+          tag="p"
+          keypath="instance.sentence.create-user-example.clickhouse.template"
+        >
+          <template #password>
+            <span class="text-red-600">YOUR_DB_PWD</span>
+          </template>
+          <template #link>
+            <a
+              class="normal-link"
+              href="https://clickhouse.com/docs/en/operations/access-rights/#access-control-usage"
+              target="__blank"
+            >
+              {{
+                $t(
+                  "instance.sentence.create-user-example.clickhouse.sql-driven-workflow"
+                )
+              }}
+            </a>
+          </template>
+        </i18n-t>
+      </template>
+      <template v-else-if="props.engineType == 'POSTGRES'">
+        <BBAttention
+          class="mb-1"
+          :style="'WARN'"
+          :title="$t('instance.sentence.create-user-example.postgres.warn')"
+        />
+        <i18n-t
+          tag="p"
+          keypath="instance.sentence.create-user-example.postgres.template"
+        >
+          <template #password>
+            <span class="text-red-600">YOUR_DB_PWD</span>
+          </template>
+        </i18n-t>
+      </template>
+      <template v-else-if="props.engineType == 'SNOWFLAKE'">
+        <i18n-t
+          tag="p"
+          keypath="instance.sentence.create-user-example.snowflake.template"
+        >
+          <template #password>
+            <span class="text-red-600">YOUR_DB_PWD</span>
+          </template>
+          <template #warehouse>
+            <span class="text-red-600">YOUR_COMPUTE_WAREHOUSE</span>
+          </template>
+        </i18n-t>
+      </template>
+      <div class="mt-2 flex flex-row">
+        <span
+          class="flex-1 min-w-0 w-full inline-flex items-center px-3 py-2 border border-r border-control-border bg-gray-50 sm:text-sm whitespace-pre"
+        >
+          {{ grantStatement(props.engineType, props.dataSourceType) }}
+        </span>
+        <button
+          tabindex="-1"
+          class="-ml-px px-2 py-2 border border-gray-300 text-sm font-medium text-control-light disabled:text-gray-300 bg-gray-50 hover:bg-gray-100 disabled:bg-gray-50 focus:ring-control focus:outline-none focus-visible:ring-2 focus:ring-offset-1 disabled:cursor-not-allowed"
+          @click.prevent="copyGrantStatement"
+        >
+          <heroicons-outline:clipboard class="w-6 h-6" />
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { defineProps, reactive, PropType } from "vue";
+import { useStore } from "vuex";
+import { toClipboard } from "@soerenmartius/vue3-clipboard";
```

Comment:
Just mention here, If we can use `useClipboard` hooks of `@vueuse/core` instead of the `toClipboard`, we can remove the dependency `soerenmartius/vue3-clipboard`

```ts
import { useClipboard } from '@vueuse/core'
```

Reply:
There are lots of components that use `soerenmartius/vue3-clipboard`, let me remove them in the next separate PR.